src/mixxx.cpp

```diff
@@ -81,10 +81,41 @@
 #include "preferences/dialog/dlgprefmodplug.h"
 #endif
 
+#ifdef Q_OS_LINUX
```

Comment:
Please replace all occurrences of `#ifdef Q_OS_LINUX` in this patch with `#if defined(Q_WS_X11)` that we already used for this purpose in other parts of Mixxx.

Reply:
I can't find any documentation for Q_WS_X11 for Qt5. Conceptually it doesn't make sense. The window system is not determined at compile, but at runtime. I think Q_OS_LINUX is correct.

Reply:
What happens if you neither have X11 nor its development headers installed? This might not happen soon, but it will happen. We use Qt as an abstraction and it doesn't matter, but here the abstractions leaks through at compile time.

Reply:
I think we can defer that until it becomes a real problem. If Qt doesn't help us when that comes, we'll have to detect the presence of X headers in our build system.